homeassistant/components/weather/meteoswiss.py

```diff
@@ -0,0 +1,188 @@
+        return False
+
+    # create condition helper
+    if DATA_CONDITION not in hass.data:
+        hass.data[DATA_CONDITION] = [str(x) for x in range(0, 200)]
+        for cond, condlst in CONDITION_CLASSES.items():
+            for condi in condlst:
+                hass.data[DATA_CONDITION][condi] = cond
+
+    add_entities([MeteoSwissWeather(meteoswiss_api, name)], True)
+
+
+class MeteoSwissWeather(WeatherEntity):
+    """Representation of MeteoSwiss data."""
+
+    def __init__(self, weather_data, name):
+        """Initialize the sensor."""
+        self._name = name
+        self._data = weather_data
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def condition(self):
+        """Return the current condition."""
+        try:
+            return self.hass.data[DATA_CONDITION][int(
+                self._data.meteoswiss['currentWeather']['icon'])]
+        except (ValueError, IndexError):
+            return STATE_UNKNOWN
+
+    @property
+    def temperature(self):
+        """Return the temperature."""
+        return float(self._data.meteoswiss['currentWeather']['temperature'])
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def pressure(self):
+        """Return the pressure."""
+        return None
+
+    @property
+    def humidity(self):
+        """Return the humidity."""
+        return None
+
+    @property
+    def visibility(self):
+        """Return the visibility."""
+        return None
+
+    @property
+    def wind_speed(self):
+        """Return the wind speed."""
+        return None
+
+    @property
+    def wind_bearing(self):
+        """Return the wind direction."""
+        return None
+
+    @property
+    def attribution(self):
+        """Return the attribution."""
+        return ATTRIBUTION
+
+    @property
+    def forecast(self):
+        """Return the forecast array."""
+        try:
+            return [
+                {
+                    ATTR_FORECAST_TIME: v['dayDate'],
+                    ATTR_FORECAST_TEMP:int(v['temperatureMax']),
+                    ATTR_FORECAST_TEMP_LOW: int(v['temperatureMin']),
+                    ATTR_FORECAST_PRECIPITATION: float(v['precipitation']),
+                    ATTR_FORECAST_CONDITION:
+                        self.hass.data[DATA_CONDITION][int(v['iconDay'])]
+                } for v in self._data.meteoswiss['forecast']]
+        except (ValueError, IndexError):
+            return STATE_UNKNOWN
+
+    def update(self):
+        """Get the latest data from MeteoSwiss and updates the states."""
+        self._data.update()
+        if not self._data.meteoswiss:
+            _LOGGER.info("Don't receive weather data from MeteoSwiss")
+            return
+
+
+class MeteoSwissData:
+    """Handle the calls to MeteoSwiss API."""
```

Comment:
Please move the data retrieval part into a separate Python Module which is available at PyPI and can be imported.

Would also allow one to create a `geo_location` platform with the warnings while using the same module.